### PR TITLE
fix feathery auth session when returning to embedded feathery form…

### PR DESCRIPTION
**Pull request by obmelvin**
…with existing stytch session

also cleans up the stytch code

### CODE REVIEW

File: src/integrations/stytch.js [hunk 1]
```diff
@@ -59,45 +59,86 @@ export function sendMagicLink(fieldVal) {
...
-    authFn = () => stytchClient.magicLinks.authenticate(token, opts);
-  } else {
-    return;
+  if (jwt && session) {
+    console.log('session emailLogin', session);
+    // This path to the user email only works for magic link oauth doesn't
+    // contain the email as part of the auth factor. I do have access to a few
+    // stytch IDs (will post in github comments). Not sure if you can use that instead?
+    const authEmail =
+      session.authentication_factors[0]?.email_factor?.email_address ??
```
Comment (obmelvin) [inline src/integrations/stytch.js:83]: This path to the user email only works for magic link oauth doesn't
contain the email as part of the auth factor. I do have access to a few
stytch IDs (will post in github comments). Not sure if you can use that instead?

File: src/utils/init.js [hunk 2]
```diff
@@ -158,8 +159,12 @@ function validateStep(formKey, trigger = true) {
   return callback(trigger);
 }
 
-function setAuthClient(client) {
+function setAuthClient(client, triggerLogin = true) {
   initState.authClient = client;
+  // NEED to handle stytch vs firebase - which emailLogin to call
+  // not sure on the best way to get session.integrations info here
+  // Is it saved somewhere I'm overlooking? should we save it to the client or init?
+  if (triggerLogin) emailLogin(defaultClient);
```
Comment (obmelvin) [inline src/utils/init.js:167]: // NEED to handle stytch vs firebase - which emailLogin to call
  // not sure on the best way to get session.integrations info here
  // Is it saved somewhere I'm overlooking? should we save it to the client or init?

File: src/integrations/utils.js [hunk 3]
```diff
@@ -51,5 +52,6 @@ export function transformQueryParamsToUrl() {
 
   const queryParams = new URLSearchParams(search);
   const redirect = queryParams.get('redirect');
-  return `${origin}${redirect ?? ''}`;
+  if (!redirect) return null;
```
Comment (obmelvin) [inline src/integrations/utils.js:55]: I think I need to update both these functions to ensure they non-redirect query params & URL hashes

Comment (bo-dun-1) [reply src/integrations/utils.js:54]: make this a ternary

File: src/integrations/stytch.js [hunk 4]
```diff
@@ -59,45 +59,82 @@ export function sendMagicLink(fieldVal) {
...
+      'oliver@feathery.io';
+    removeStytchQueryParams();
+    return featheryClient
+      .submitAuthInfo({
+        authId: session.user_id,
+        authEmail,
+        is_stytch_template_key: config.is_stytch_template_key
+      })
+      .error(() => (authSent = false));
```
Comment (bo-dun-1) [inline src/integrations/stytch.js:88]: isn't it .catch?

File: src/integrations/stytch.js [hunk 5]
```diff
@@ -59,45 +59,82 @@ export function sendMagicLink(fieldVal) {
...
+
+    return authFn()
+      .then((result) =>
+        featheryClient
+          .submitAuthInfo({
+            authId: result.user.user_id,
+            authEmail: result.user.emails[0].email,
+            is_stytch_template_key: config.is_stytch_template_key
+          })
+          .error(() => (authSent = false))
```
Comment (bo-dun-1) [inline src/integrations/stytch.js:114]: .catch?

**Review by bo-dun-1 — CHANGES_REQUESTED** [review]

File: src/integrations/stytch.js (hunk 4 above)
Comment (obmelvin) [reply src/integrations/stytch.js]: lol, yes

**Review by bo-dun-1 — APPROVED** [review]

**Pull request merged**